gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/oauth2/OAuth2SsoConfig.groovy

```diff
@@ -134,4 +137,15 @@ class OAuth2SsoConfig extends OAuth2SsoConfigurerAdapter {
       }
     }
   }
+
+  /**
+   * Use this class to specify how to map fields from the userInfoUri response to what's expected to be in the User.
+   */
+  @Component
+  @ConfigurationProperties("spring.oauth2.userInfoMapping")
+  static class UserInfoMapping {
```

Comment:
I think introducing a `userId` for a unique identifier might be useful. Not all OAuth providers use `email` as identifier (for instance, GitHub does not, the `login` is the identifier even you cannot create 2 GitHub accounts with the same email.
Also, for the authZ part, to retrieve a GitHub user memberships, it needs the `login` as input (and not the email).
So I think we could have a `userId` field, (filled with `login` for Github, `email` for Google), and the `UserRolesProvider` (https://github.com/spinnaker/gate/blob/google-oauth/gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/rolesprovider/UserRolesProvider.groovy#L27) functions might user `userId` as inputs.
This way it would more decoupled of how an OAuth provider identifies its users.